enhancements/microshift/microshift-default-csi-plugin.md

```diff
@@ -0,0 +1,345 @@
+    metadata:
+      name: my-pod
+    spec:
+      containers:
+      - name: nginx
+        image: nginx
+        command: ["/usr/bin/sh", "-c"]
+        args: ["sleep", "1h"]
+        volumeMounts:
+        - mountPath: /mnt
+          name: my-volume
+      volumes:
+        - name: my-volume
+          persistentVolumeClaim:
+            claimName: my-lv-pvc
+    EOF
+    ```
+
+**Maintainer Driven Rebase**
+
+Because MicroShift is based on OCP, it must ensure parity between its cluster components and those of OCP, by version.
+OCP records cluster component image digests in the ocp-release image, which is versioned according to its corresponding
+OCP version.  ODF maintains a similar release image called ocs-release, which maps ODF-LVM image digests to OCP versions.
+The workflow is automated and documented 
+[here](https://github.com/openshift/microshift/blob/9dfddbd6801098c96bbd7db887c2844cad177c69/docs/rebase.md).
+
+#### Variation
+
+**Start MicroShift with a user-provided ODF-LVM configuration**
+
+> After the device owner has logged into the device ... 
+
+1. The device owner will create a file called `/etc/microshift/lvmd.yaml`
+1. The device owner will define the lvmd config, according to its 
+[documentation](https://github.com/red-hat-storage/topolvm/blob/main/docs/lvmd.md)
+
+   _For example:_
+   
+    ```shell
+    $ cat <<EOF > /etc/microshift/lvmd.yaml
+    socket-name: 
+    deviceClasses:
+    - default: true
+      name: "user-provided-config"
+      volume-group: rhel
+      spare-gb: 10 # in GB
+    socket-name: /run/lvmd/lvmd.sock # socket-name must always be /run/lvmd/lvmd.sock
+    ```
+
+1. The device owner will start the microshift service, and proceed to validate the platform as described in **Validate 
+ODF-LVM function**
+1. The device owner will validate the system, following workflow **Validate ODF-LVM function**
+
+**Start MicroShift with a *malformed* user-provided ODF-LVM configuration**
+
+> After the device owner has logged into the device ...
+
+1. The device owner creates an incorrect lvmd config file under /etc/microshift/lvmd.config
+
+    _For example:_
+    ```shell
+    [root@ushift ~]# cat <<EOF > /etc/microshift/lvmd.yaml
+    socket-name: 
+    deviceClasses:
+    - default: true
+      name: "user-provided-config"
+      volume-group: non_existent_volume_group
+      spare-gb: 10 # in GB
+    socket-name: /run/lvmd/lvmd.sock
+    ```
+**Correct a _malformed_ user-provided ODF-LVM configuration and restart MicroShift**
+
+> After the device owner has logged into the device ...
+
+1. The device owner will edit the /etc/microshift/lvmd.yaml file to correct the error
+1. The device owner will restart MicroShift: `systemctl restart microshift`
+1. The device owner will validate the system, following workflow **Validate ODF-LVM function**
+
+### API Extensions
+
+**Cluster CRDs**
+
+ODF-LVM represents logical volumes with a CRD called `logicalvolumes.topolvm.cybozu.com`.  This CRD is installed on the 
```

Comment:
the CRD domain will change before we release ODF 4.12
@nbalacha - what will be the new domain name?

Reply:
OK, that's convenient for us. @copejon we should make sure we track that change closely and try to pull in the version with the new name. 

We may still need to provide a way to set the environment variable because our goal is to support deploying workloads configured the same way on OCP and MicroShift, but if we can avoid a breaking change down the road that would be ideal.

Reply:
I agree.  Any current users would be a risk of data loss otherwise.

Reply:
@mulbc and @nbalacha , what will the default setting be in in LVMO?

Reply:
Starting with 4.12 we want to use the new domain name - right @nbalacha ?
I think further down in the document you actually talk about this, but we should probably use the new name here already, since the doc will be used after ODF 4.12 is released? Or we omit the domain name

Reply:
Fair point. I've modified the doc to reflect the new group.

Reply:
@mulbc As for the environment variable, how will the LMVO handle legacy clusters?

Reply:
AFAIK they will still work, but we don't support that. @nbalacha would probably know these things

Reply:
Even so, the proposal should probably use `topolvm.io` to prevent any confusion it might create after the change over happens.